### PR TITLE
Add Optional Skip Module

### PR DESCRIPTION
**Pull request by acidmanifesto**
**Changes proposed**:

Add optional skip DK starter zone quest line thru gossip option with lich king on creation

**Target branch(es)**: 335

### CODE REVIEW

File: sql/custom/world/world.DKS.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-- npc id lich king
+-- Original npcflag was 2
+-- no previous scriptname
+UPDATE `creature_template` SET `npcflag`='3, `ScriptName`='npc_tc_skip_lich' WHERE  `entry`=25462;
```
Comment (Aokromes) [inline sql/custom/world/world.DKS.sql]: missing newline.

Comment (acidmanifesto) [reply sql/custom/world/world.DKS.sql]: > missing newline.

New line? im not understanding.
what this does is makes the first lich king npc u see have a additional  gossip for the option al skip.
image

Comment (acidmanifesto) [reply sql/custom/world/world.DKS.sql]: > missing newline.

I believe i was informed by what you ment. I went ahead and added a new line. So we wait for travis to finish now. @Aokromes

**Comment by acidmanifesto** [comment]
@Aokromes ready for review.

File: sql/custom/world/.gitignore [hunk 2]
```diff
@@ -1 +0,0 @@
-*.sql
```
Comment (jackpoz) [inline sql/custom/world/.gitignore]: you can keep this and still force adding the sql below. I would prefer to not have this .gitignore modified

**Review by jackpoz — CHANGES_REQUESTED** [review]
Looks good, just the .gitignore change I would like not to be in this pr

File: sql/custom/world/.gitignore (hunk 2 above)
Comment (acidmanifesto) [reply sql/custom/world/.gitignore]: Thank you. I went ahead and re-included the gitignore.

File: sql/custom/world/.gitignore [hunk 3]
```diff
@@ -1 +1 @@
...
+*.sql
```
Comment (jackpoz) [inline sql/custom/world/.gitignore]: You might have to add a new line to restore the exact content as before, so git will see it as unmodified

File: src/server/scripts/Custom/SkipDK.cpp [hunk 4]
```diff
@@ -158,8 +162,145 @@ class Trinitycore_skip_deathknight : public PlayerScript
...
+                {
+                case LOCALE_koKR: localizedEntry = LOCALE_LICHKING_1; break;
+                case LOCALE_frFR: localizedEntry = LOCALE_LICHKING_2; break;
+                case LOCALE_deDE: localizedEntry = LOCALE_LICHKING_3; break;
+                case LOCALE_zhCN: localizedEntry = LOCALE_LICHKING_4; break;
+                case LOCALE_zhTW: localizedEntry = LOCALE_LICHKING_5; break;
+                case LOCALE_esES: localizedEntry = LOCALE_LICHKING_6; break;
+                case LOCALE_esMX: localizedEntry = LOCALE_LICHKING_7; break;
+                case LOCALE_ruRU: localizedEntry = LOCALE_LICHKING_8; break;
+                case LOCALE_enUS: default: localizedEntry = LOCALE_LICHKING_0;
```
Comment (jackpoz) [inline src/server/scripts/Custom/SkipDK.cpp]: Please add a break at the end, even if it's the last case

File: src/server/scripts/Custom/SkipDK.cpp [hunk 5]
```diff
@@ -158,8 +162,145 @@ class Trinitycore_skip_deathknight : public PlayerScript
...
+                player->EquipNewItem(EQUIPMENT_SLOT_TRINKET2, 38674, true);//Soul Harvester's Charm
+                player->EquipNewItem(EQUIPMENT_SLOT_FINGER1, 38671, true);//Valanar's Signet Ring
+                player->EquipNewItem(EQUIPMENT_SLOT_FINGER2, 38672, true);// Keleseth's Signet Ring
+                player->AddItem(39320, true);//Sky Darkener's Shroud of Blood
+                player->AddItem(38664, true);//Sky Darkener's Shroud of the Unholy
+                player->AddItem(39322, true);//Shroud of the North Wind
+                player->AddItem(38632, true);//Greatsword of the Ebon Blade
+                player->AddItem(6948, true);//Hearthstone
+                player->AddItem(38707, true);//Runed Soulblade
+                player->AddItem(40483, true);//Insignia of the Scourge
```
Comment (jackpoz) [inline src/server/scripts/Custom/SkipDK.cpp:260]: What happens if the player has no free space ? Are items lost or emailed automatically ? Is that good enough or should it be modified ? In this PR or in another PR ?

Comment (acidmanifesto) [reply src/server/scripts/Custom/SkipDK.cpp:260]: There is plenty of space as the DK class starts off with 4 bags and the starter bag. Only class in wotlk that starts off with five bags total. and the items do not max out the inventory space of the main player bag. The items will not be lost out side of the basic armor upgrade that every DK player will have when they are done with the starter area. The option to optional skip will only be unique to the first lich king npc which will only be present for about 3 quests, after that the Option to skip the DK becomes absent from the player. This is as intended by design.

**Review by jackpoz — APPROVED** [review]

**Pull request merged**